lib/puppet/provider/package/windows/exe_package.rb

```diff
@@ -55,7 +57,26 @@ def match?(resource)
     end
 
     def self.install_command(resource)
-      munge(resource[:source])
+      if Puppet::FileSystem.exist?(resource[:source])
+        munge(resource[:source])
+      else
+        begin
+          Puppet.debug ("The source %{resource_source} does not exist locally. Using it as URI ") % { resource_source: resource[:source]}
+          uri = URI.parse(resource[:source])
+        rescue URI::InvalidURIError => e
+          Puppet.debug(_("URI is invalid: %{detail}") % { detail: e.message}, e)
+        rescue => detail
+          raise Puppet::Error.new(_("Error when parsing URI %{detail}") % { detail: detail.message}, detail)
+        end
```

Comment:
I think we want to raise in both rescue cases. Maybe just drop the `rescue URI::InvalidURIError` clause?